milvus/provider/provider.py

```diff
@@ -1,62 +1,52 @@
-from flask import current_app as app
 import logging
 
-import cohere
-from pymilvus import connections, Collection
-
+from flask import current_app as app
+from .client import get_cohere_client, get_milvus_client
 
 logger = logging.getLogger(__name__)
-milvus_connection = None
-cohere_client = None
 
 
-def search(query):
-    global milvus_connection
-    global cohere_client
-
-    if not cohere_client:
-        assert (
-            apikey := app.config.get("COHERE_APIKEY")
-        ), "MILVUS_COHERE_APIKEY env var must be set"
-        cohere_client = cohere.Client(apikey)
-    if not milvus_connection:
-        milvus_connection = connections.connect(
-            alias="default",
-            host=app.config["CLUSTER_HOST"],
-            port=app.config["CLUSTER_PORT"],
+def serialize_results(data, mappings={}):
+    """
+    Serialize a list of dictionaries by transforming keys based on provided mappings
+    and converting values to strings.
+
+    Parameters:
+    - data (list): A list of dictionaries to be serialized.
+    - mappings (dict): A dictionary specifying key mappings for transformation.
+
+    Returns:
+    list: A serialized list of dictionaries with transformed keys and string-converted values.
+    """
+    serialized_data = list(
+        map(
+            lambda item: {
+                k
+                if k not in mappings
+                else mappings[k]: ", ".join(str(vl) for vl in v)
+                if isinstance(v, list)
+                else str(v)
+                for k, v in item.items()
```

Comment:
Could we break this out into a proper function instead of a lambda? Finding the `if ... else` hard to follow here

Reply:
Thank you. Done